### PR TITLE
Handle output L0 stream attributes

### PR DESCRIPTION
**Pull request by bmerry**
This is to handle the new attributes written by ingest
(ska-sa/katsdpingest#194) and filewriter (ska-sa/katsdpdata#71), which
allow for
- The file to contain either the continuum or spectral L0 stream
  (disambiguated by an HDF5 attribute on the /Data group);
- The file to contain only a subset of channels, and thus have separate
  n_chans, bandwidth, centre frequency to the CBF.

An area that will need some review is the precedence of the different
ways of determining centre frequency. Currently I've made it trust the
SDP-written value (e.g. sdp_l0_center_freq) over anything else, because
values determined from the band don't take account of channel subsetting
in ingest. However, that in turn means trusting the CAM centre
frequency, which is probably not correct for Ku band. I propose that the
right way to handle that is to fix CAM to be Ku band aware before trying
to update RTS with the latest ingest and filewriter.

### CODE REVIEW

**Comment by ludwigschwardt** [comment]
When are we rolling this out to site?

File: katdal/h5datav3.py [hunk 1]
```diff
@@ -482,17 +482,17 @@ def register_telstate_sensor(name, obj):
...
         stream_centre_freq = self._get_telstate_stream_attr('center_freq')
         if stream_centre_freq is not None:
             spw_params['centre_freq'] = stream_centre_freq
-        # Get channel width from original CBF parameters
+        # Get channel width from original CBF / SDP parameters
         spw_params['channel_width'] = bandwidth / num_chans
         # Continue with different channel count, but invalidate centre freq (keep channel width though)
         if num_chans != self._vis.shape[1]:
-            logger.warning('Number of channels received from ingest (%d) differs '
+            logger.warning('Number of channels received from CBF / ingest (%d) differs '
```
Comment (bmerry) [inline katdal/h5datav3.py]: Why this change? When does a v3 file ever contain data received from CBF?

File: katdal/h5datav3.py [hunk 2]
```diff
@@ -579,6 +579,7 @@ def _get_telstate_attr(self, key, default=None, no_unpickle=()):
         try:
             value = self.file['TelescopeState'].attrs[key]
             if value in no_unpickle:
+                # Telstate attributes were serialised via str() until 2016-11-29
```
Comment (bmerry) [inline katdal/h5datav3.py]: Maybe merge this comment into the last sentence of the docstring.

File: katdal/h5datav3.py (hunk 1 above)
Comment (ludwigschwardt) [reply katdal/h5datav3.py]: I was thinking of the source of the information - ingest just passed CBF spead attrs through in the past. Now it can also modify them.

**Comment by bmerry** [comment]
> When are we rolling this out to site?

Do you mean this PR, or the code that produces these attributes? The former you could deploy ASAP, since it (should be) backwards compatible. I'll try to get the latter out this month, although it affects multiple images so will need some coordination.

**Comment by ludwigschwardt** [comment]
The latter, yes.

File: katdal/h5datav3.py (hunk 1 above)
Comment (ludwigschwardt) [reply katdal/h5datav3.py]: Improved.

File: katdal/h5datav3.py (hunk 2 above)
Comment (ludwigschwardt) [reply katdal/h5datav3.py]: Done.

**Comment by bmerry** [comment]
Latest updates look good. It's your package, so I'll let you merge when it's convenient for you.
